src/main/java/com/fasterxml/jackson/core/util/BufferRecyclerPool.java

```diff
@@ -0,0 +1,14 @@
+package com.fasterxml.jackson.core.util;
+
+public class BufferRecyclerPool {
+
+    private static final ObjectPool<BufferRecycler> pool = ObjectPool.newObjectPool(new BufferRecycler()::withPool);
+
+    public static BufferRecycler acquireBufferRecycler() {
+        return pool.acquire();
+    }
```

Comment:
Any chance that you could keep a count of acquired and released recyclers?

Could be useful in debugging issues where pools are not released.

Reply:
I added a `DebugPoolDecorator` that can be optionally enabled to this purpose.